ui/test/nuxeo-lock-toggle-button.test.js

```diff
@@ -14,82 +14,339 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+      await flush();
+      expect(dom(button.root).querySelector('.action')).to.be.null;
+    });
+
+    test("Should not be visible when document type is 'Root'", async () => {
+      button.document = {
+        ...lockedDocument,
+        type: 'Root',
+      };
+      await flush();
+      expect(dom(button.root).querySelector('.action')).to.be.null;
+    });
+
+    test('Should be visible otherwise', async () => {
+      button.document = lockedDocument;
+      await flush();
+
+      const actionDiv = dom(button.root).querySelector('.action');
+      expect(actionDiv).to.not.be.null;
+      expect(actionDiv.querySelector('.label')).to.not.be.null;
+      expect(actionDiv.querySelector('paper-icon-button')).to.not.be.null;
+    });
+
+    test('Should have corresponding information when visible and locked', async () => {
+      button.document = lockedDocument;
+      await flush();
+
+      const actionDiv = dom(button.root).querySelector('.action');
+      expect(actionDiv).to.not.be.null;
+
+      const label = actionDiv.querySelector('.label').textContent.trim();
+      expect(label)
+        .to.be.a('string')
+        .and.to.equal('lockToggleButton.tooltip.unlock');
+
+      const icon = actionDiv.querySelector('paper-icon-button');
+      expect(icon).to.not.be.null;
+      expect(icon.icon).to.equal('nuxeo:lock');
+    });
+
+    test('Should have corresponding information when visible and unlocked', async () => {
+      button.document = {
+        ...lockedDocument,
+        lockCreated: null,
+        lockOwner: null,
       };
-      element = await fixture(
-        html`
-          <nuxeo-lock-toggle-button .document=${doc}></nuxeo-lock-toggle-button>
-        `,
-      );
+      await flush();
+
+      const actionDiv = dom(button.root).querySelector('.action');
+      expect(actionDiv).to.not.be.null;
+
+      const label = actionDiv.querySelector('.label').textContent.trim();
+      expect(label)
+        .to.be.a('string')
+        .and.to.equal('lockToggleButton.tooltip.lock');
+
+      const icon = actionDiv.querySelector('paper-icon-button');
+      expect(icon).to.not.be.null;
+      expect(icon.icon).to.equal('nuxeo:unlock');
+    });
+  });
+
+  suite('Server Responds with Exceptions', () => {
+    setup(async () => {
+      sinon.spy(button.$.opUnlock, 'execute');
+      sinon.spy(button.$.opLock, 'execute');
+    });
+
+    test('Should trigger notify event when trying to unlock document and permissions error is received', async () => {
       server.respondWith('POST', '/api/v1/automation/Document.Unlock', [
-        200,
+        403,
         { 'Content-Type': 'application/json' },
-        '{"entity-type": "document","uid": "1"}',
+        JSON.stringify({
+          'entity-type': 'exception',
+          message: 'FORBIDDEN',
+          status: 403,
+        }),
       ]);
+
+      button.document = lockedDocument;
+      tap(button);
+      expect(button.$.opUnlock.execute.calledOnce).to.be.equal(true);
```

Comment:
We should use the short version for consistency, since we're already using it this way in this file:
```suggestion
      expect(button.$.opUnlock.execute.calledOnce).to.be.true;
```

Reply:
Ops! Of course!